bin/gwdetchar-lasso-correlation

```diff
@@ -285,97 +345,316 @@ def process_channel(input_,):
            
         
         # Create scaled, sign-corrected, and overlayed timeseries
-        tsscaled = ts.detrend()
-        tsrms = numpy.sqrt(sum(tsscaled**2.0)/len(tsscaled))
-        if args.trend_type == 'minute':
-            tsscaled *= (rangerms / tsrms)
-            if lassocoef < 0:
-                tsscaled *= -1
-
-        plot = TimeSeriesPlot(rangescaled, tsscaled,
-                              figsize=[12, 6])
-        plot.subplots_adjust(*p2)
-        ax = plot.gca()
-        ax.set_xlim(start, end)
-        ax.set_epoch(start)
-        ax.set_ylabel('Scaled amplitude [arbitrary units]')
-        ax.legend(loc='best')
+
+	tsscaled = scale(ts.value)
+	if lassocoef < 0:
+            tsscaled = numpy.negative(tsscaled)
+	fig = plt.figure(figsize=(12,6))
+	plt.plot(scale(rangets.value), label=rangechannel.replace("_","\_"))
+	plt.plot(tsscaled, label=chan.replace("_","\_"))
+	plt.xlabel('Time [min]')
+	plt.ylabel('Scaled amplitude [arbitrary units]')
+	plt.legend(loc='best')
+	plt.tight_layout()
+
         plot2 = '%s_COMPARISON-%s.png' % (channelstub, gpsstub)
         try:
-            plot.save(plot2)
+            plt.savefig(plot2)
         except (IOError, IndexError):
-            plot.save(plot2)
+            plt.savefig(plot2)
         except RuntimeError as e:
             if 'latex' in str(e).lower():
-                plot.save(plot2)
+                plt.savefig(plot2)
+            else:
+                raise
+        plt.close()
+
+        #plot scatter plots
+        rangeColor="red"
+        plotHeight=6
+        plotWidth=12
+
+        tsCopy=ts.reshape(-1,1)
+        rangetsCopy=rangets.reshape(-1,1)
+        rangeReg = linear_model.LinearRegression()
+        rangeReg.fit(tsCopy, rangetsCopy)
+        rangeFit = rangeReg.predict(tsCopy)
+        scatPlot=gwplot()
+        axes = scatPlot.gca()
+        axes.set_xlabel(chan.replace("_","\_") + " [Channel units]", labelpad=30)
+        axes.set_ylabel("Sensitive range [Mpc]")
+        axes.text(.9,.1,"r = " + str('{0:.2}'.format(pcorr)), verticalalignment='bottom', horizontalalignment='right', transform=axes.transAxes, color="black", size=20, bbox=dict(boxstyle='square', facecolor='white', alpha=.75, edgecolor="black"))
+        scatPlot.add_scatter(ts, rangets, color=rangeColor)
+        scatPlot.add_line(ts, rangeFit, color="black")
+        scatPlot.set_figheight(plotHeight)
+        scatPlot.set_figwidth(plotWidth)
+	scatPlot.tight_layout()
```

Comment:
@macedo22, does this line have tabs? the indentation looks inconsistent?

Reply:
Yes, it seems a few tabs were still hidden in there. I will remove them.